### PR TITLE
Watermarking

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: boxsdk/object/folder.py [hunk 1]
```diff
@@ -382,3 +382,39 @@ def delete(self, recursive=True, etag=None):
         :raises: :class:`BoxAPIException` if the specified etag doesn't match the latest version of the folder.
         """
         return super(Folder, self).delete({'recursive': recursive}, etag)
+
+
+
+    def get_watermark(self):
+        """
+        Return the watermark info for a Box folder
```
Comment (SajadMirzaei) [inline boxsdk/object/folder.py]: doc missing return type

File: boxsdk/object/folder.py [hunk 2]
```diff
@@ -382,3 +382,39 @@ def delete(self, recursive=True, etag=None):
...
+        url = '{0}/folders/{1}/{2}'.format(API.BASE_API_URL, self.object_id, 'watermark')
+        box_response = self._session.get(url, expect_json_response=False)
+        return box_response.json()
+
+
+
+    def apply_watermark(self):
+        """
+        Apply watermark on a Box folder
+        """
```
Comment (SajadMirzaei) [inline boxsdk/object/folder.py]: doc missing return type

File: boxsdk/object/folder.py [hunk 3]
```diff
@@ -382,3 +382,39 @@ def delete(self, recursive=True, etag=None):
...
+        }
+        box_response = self._session.put(url, data=json.dumps(body_attributes))
+        return box_response.json()
+
+
+
+    def delete_watermark(self):
+        """
+        Return the watermark info for a Box folder
+        """
```
Comment (SajadMirzaei) [inline boxsdk/object/folder.py]: doc missing return type

File: boxsdk/object/file.py [hunk 4]
```diff
@@ -289,3 +290,38 @@ def get_shared_link_download_url(
...
         )
         return item.shared_link['download_url']
+
+
+
+    def get_watermark(self):
+        """
+        Return the watermark info for a Box file
+        """
+        url = '{0}/files/{1}/{2}'.format(API.BASE_API_URL, self.object_id, 'watermark')
```
Comment (SajadMirzaei) [inline boxsdk/object/file.py]: 'watermark' can be put back into the actual string instead of formatting.

File: boxsdk/object/file.py [hunk 5]
```diff
@@ -289,3 +290,38 @@ def get_shared_link_download_url(
...
+        box_response = self._session.get(url, expect_json_response=False)
+        return box_response.json()
+
+
+
+    def apply_watermark(self):
+        """
+        Apply watermark on a Box file
+        """
+        url = '{0}/files/{1}/{2}'.format(API.BASE_API_URL, self.object_id, 'watermark')
```
Comment (SajadMirzaei) [inline boxsdk/object/file.py]: same

File: boxsdk/object/file.py [hunk 6]
```diff
@@ -289,3 +290,38 @@ def get_shared_link_download_url(
...
+        box_response = self._session.put(url, data=json.dumps(body_attributes))
+        return box_response.json()
+
+
+
+    def delete_watermark(self):
+        """
+        Return the watermark info for a Box file
+        """
+        url = '{0}/files/{1}/{2}'.format(API.BASE_API_URL, self.object_id, 'watermark')
```
Comment (SajadMirzaei) [inline boxsdk/object/file.py]: same

File: boxsdk/object/folder.py [hunk 7]
```diff
@@ -382,3 +382,39 @@ def delete(self, recursive=True, etag=None):
         :raises: :class:`BoxAPIException` if the specified etag doesn't match the latest version of the folder.
         """
         return super(Folder, self).delete({'recursive': recursive}, etag)
+
+
```
Comment (SajadMirzaei) [inline boxsdk/object/folder.py]: Too many blank lines. (Same in other added functions)

File: boxsdk/object/file.py [hunk 8]
```diff
@@ -292,6 +293,51 @@ def get_shared_link_download_url(
...
+            'watermark': {
+                'imprint': 'default'
+            }
+        }
+        box_response = self._session.put(url, data=json.dumps(body_attributes))
+        return box_response.json()
+
+    def delete_watermark(self):
+        """
+        Return the watermark info for a Box file
```
Comment (SajadMirzaei) [inline boxsdk/object/file.py]: It deletes the watermark, isn't it?

File: boxsdk/object/folder.py [hunk 9]
```diff
@@ -419,3 +419,47 @@ def delete(self, recursive=True, etag=None):
...
+            'watermark': {
+                'imprint': 'default'
+            }
+        }
+        box_response = self._session.put(url, data=json.dumps(body_attributes))
+        return box_response.json()
+
+    def delete_watermark(self):
+        """
+        Return the watermark info for a Box folder
```
Comment (SajadMirzaei) [inline boxsdk/object/folder.py]: delete

File: test/unit/object/test_file.py [hunk 10]
```diff
@@ -277,6 +277,42 @@ def test_get_shared_link_download_url(
...
+
+    watermark = test_file.apply_watermark()
+    mock_box_session.put.assert_called_once_with(expected_url, data='{"watermark": {"imprint": "default"}}')
+    assert watermark['created_at'] == created_at
+    assert watermark['modified_at'] == modified_at
+
+
+def test_delete_watermark(mock_box_session, test_file):
+    expected_url = '{0}/files/{1}/watermark'.format(API.BASE_API_URL, test_file.object_id)
+    test_file.delete_watermark()
```
Comment (SajadMirzaei) [inline test/unit/object/test_file.py]: The return value can be asserted here (At least the type of it)

File: test/unit/object/test_folder.py [hunk 11]
```diff
@@ -287,6 +287,42 @@ def test_preflight(test_folder, mock_object_id, mock_box_session):
...
+
+    watermark = test_folder.apply_watermark()
+    mock_box_session.put.assert_called_once_with(expected_url, data='{"watermark": {"imprint": "default"}}')
+    assert watermark['created_at'] == created_at
+    assert watermark['modified_at'] == modified_at
+
+
+def test_delete_watermark(mock_box_session, test_folder):
+    expected_url = '{0}/folders/{1}/watermark'.format(API.BASE_API_URL, test_folder.object_id)
+    test_folder.delete_watermark()
```
Comment (SajadMirzaei) [inline test/unit/object/test_folder.py]: Same

File: test/unit/object/test_folder.py [hunk 12]
```diff
@@ -287,6 +287,43 @@ def test_preflight(test_folder, mock_object_id, mock_box_session):
...
+    mock_box_session.put.assert_called_once_with(expected_url, data='{"watermark": {"imprint": "default"}}')
+    assert watermark['created_at'] == created_at
+    assert watermark['modified_at'] == modified_at
+
+
+def test_delete_watermark(mock_box_session, test_folder):
+    mock_box_session.delete.return_value.ok = True
+    expected_url = '{0}/folders/{1}/watermark'.format(API.BASE_API_URL, test_folder.object_id)
+    test_folder.delete_watermark()
+    mock_box_session.delete.assert_called_once_with(expected_url, expect_json_response=False)
```
Comment (SajadMirzaei) [inline test/unit/object/test_folder.py]: The same assertion for the file can be done here too

**Pull request closed without merging**